### PR TITLE
Add e2e test job for sink-resource-ci

### PR DESCRIPTION
**Pull request by xiujuanx**
Fix issue: https://jira.eng.vmware.com/browse/PKS-4257

This PR adds e2e test job for sink-resource pipeline. It had been verified: https://pks.ci.cf-app.com/teams/pks-sink/pipelines/sink-resources/jobs/e2e-test/builds/4. By the way, although this job failed, it's none of pipeline's business. It's related to the test cases.

### CODE REVIEW

File: tasks/sink-e2e/task [hunk 1]
```diff
@@ -1,18 +1,80 @@
...
+pushd fluent-bit-out-syslog
+  fluent_bit_sha=\$(git rev-parse HEAD)
+popd
+echo "\${fluent_bit_sha}"
+
+pushd sink-resources
+  sink_resources_sha=\$(git rev-parse HEAD)
+  echo "\${sink_resources_sha}"
+  ./hack/generate-pks-manifest.sh "\${sink_resources_sha}" "\${fluent_bit_sha}" | kubectl apply -f -
+  kubectl delete validatingwebhookconfigurations validator.pksapi.io
```
Comment (wangyuan0916) [inline tasks/sink-e2e/task:69]: remove this, this is for debugging

File: tasks/sink-e2e/task [hunk 2]
```diff
@@ -1,18 +1,80 @@
...
+  # Waiting for all resources are reday
+  sleep 5m
+  # Run CRD test
+  ./tests/test.sh
+  # Run e2e test cases
+  go test -tags=e2e ./tests/e2e/... -v -count=1 -timeout=20m
```
Comment (wangyuan0916) [inline tasks/sink-e2e/task:75]: 60m

File: tasks/sink-e2e/task (hunk 1 above)
Comment (xiujuanx) [reply tasks/sink-e2e/task:67]: Done!

File: tasks/sink-e2e/task (hunk 2 above)
Comment (xiujuanx) [reply tasks/sink-e2e/task]: Done!

**Review by wangyuan0916 — APPROVED** [review]

**Pull request merged**